pkg/util/bookkeepercluster_test.go

```diff
@@ -0,0 +1,304 @@
+	Context("IsOrphan", func() {
+		var result1, result2, result3, result4 bool
+		BeforeEach(func() {
+
+			result1 = IsOrphan("bookie-4", 3)
+			result2 = IsOrphan("bookie-2", 3)
+			result3 = IsOrphan("bookie", 1)
+			result4 = IsOrphan("bookie-1ab", 1)
+		})
+		It("should return true for result2", func() {
+			Ω(result1).To(Equal(true))
+		})
+		It("should return false for result1", func() {
+			Ω(result2).To(Equal(false))
+		})
+		It("should return false for result3", func() {
+			Ω(result3).To(Equal(false))
+		})
+		It("should return false for result4", func() {
+			Ω(result4).To(Equal(false))
+		})
+	})
+	Context("HealthcheckCommand()", func() {
+
+		out := HealthcheckCommand(1234)
+		It("should not be nil", func() {
+			Ω(len(out)).ShouldNot(Equal(0))
+		})
+
+	})
+	Context("Min()", func() {
+
+		It("Min should be 10", func() {
+			Ω(Min(10, 20)).Should(Equal(int32(10)))
+		})
+		It("Min should be 20", func() {
+			Ω(Min(30, 20)).Should(Equal(int32(20)))
+		})
+
+	})
+	Context("ContainsStringWithPrefix", func() {
+		var result, result1 bool
+		BeforeEach(func() {
+			opts := []string{
+				"-Xms512m",
+				"-XX:+ExitOnOutOfMemoryError",
+			}
+
+			result = ContainsStringWithPrefix(opts, "-Xms")
+			result1 = ContainsStringWithPrefix(opts, "-abc")
+		})
+		It("should return true for result", func() {
+			Ω(result).To(Equal(true))
+
+		})
+		It("should return false for result1", func() {
+			Ω(result1).To(Equal(false))
+		})
+
+	})
+	Context("RemoveString", func() {
+		var result bool
+		BeforeEach(func() {
+			opts := []string{
+				"abc-test",
+				"test1",
+			}
+			opts = RemoveString(opts, "abc-test")
+			result = ContainsStringWithPrefix(opts, "abc")
+
+		})
+		It("should return false for result", func() {
+			Ω(result).To(Equal(false))
+
+		})
+
+	})
+	Context("GetStringWithPrefix", func() {
+		var out, out1 string
+		BeforeEach(func() {
+			opts := []string{
+				"abc-test",
+				"test1",
+			}
+			out = GetStringWithPrefix(opts, "abc")
+			out1 = GetStringWithPrefix(opts, "bk")
+
+		})
+		It("should return string with prefix", func() {
+			Ω(out).To(Equal("abc-test"))
+
+		})
+		It("should return empty string", func() {
+			Ω(out1).To(Equal(""))
+		})
+	})
+	Context("GetClusterExpectedSize", func() {
+		var replicas int
+		var bk *v1alpha1.BookkeeperCluster
+		BeforeEach(func() {
+			bk = &v1alpha1.BookkeeperCluster{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "default",
+				},
+			}
+			bk.WithDefaults()
+			replicas = GetClusterExpectedSize(bk)
+		})
+
+		It("should return correct replica count", func() {
+			Ω(replicas).To(Equal(3))
+		})
+	})
+	Context("BookkeeperImage", func() {
+		var image string
+		var bk *v1alpha1.BookkeeperCluster
+		BeforeEach(func() {
+			bk = &v1alpha1.BookkeeperCluster{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "default",
+				},
+			}
+			bk.WithDefaults()
+			image = BookkeeperImage(bk)
+		})
+
+		It("should return correct image", func() {
+			Ω(image).To(Equal("pravega/bookkeeper:0.6.1"))
```

Comment:
This default `pravega/bookkeeper:0.6.1` might change and we would need to alter this test every time

Reply:
Fixed by setting image in version field